va/va_vpp.h

```diff
@@ -1447,6 +1449,75 @@ typedef struct _VAProcFilterCap3DLUT {
     uint32_t            va_reserved[VA_PADDING_HIGH];
 } VAProcFilterCap3DLUT;
 
+/**
+ *  \brief Video Processing Super Resolution Predefined Mode.
+ *
+ *  Super Resolution is the process of recovering a High Resolution(HR) image/video
+ *  from a given Low Resolution(LR) image/video. A few modes are predefined for
+ *  different scenarios.
+ */
+typedef enum _VAProcSuperResolutionMode {
+    /**
+     * \brief Super Resolution Default Mode.
+     */
+    VAProcSuperResolutionDefault = 0,
+    /**
+     * \brief Super Resolution Conference Mode.
+     * In this mode, Super Resolution is optimized for video conference scenario.
+     */
+    VAProcSuperResolutionConference,
+    /**
+     * \brief Super Resolution Camera Mode.
+     * In this mode, Super Resolution is optimized for camera scenario.
+     */
+    VAProcSuperResolutionCamera,
+    /**
+     * \brief Number of Super Resolution Mode.
+     */
+    VAProcSuperResolutionCount,
+} VAProcSuperResolutionMode;
+
+/**
+ *  \brief Super Resolution(SR) filter parametrization.
+ */
+typedef struct _VAProcFilterParameterBufferSuperResolution {
+    /**
+     * \brief Filter type. Shall be set to #VAProcFilterSuperResolution.
+     */
+    VAProcFilterType                 type;
+    /**
+     * \brief Super Resolution mode, should be one of VAProcSuperResolutionXXX
+     */
+    VAProcSuperResolutionMode        mode;
+    /**
+     *\brief reserved bytes for future use, must be zero
+     */
+    uint32_t                         va_reserved[VA_PADDING_HIGH];
+} VAProcFilterParameterBufferSuperResolution;
+
+/**
+ * \brief Capabilities specification for the Super Resolution filter.
+ *
+ * Super resolution filter capabilities can be checked with vaQueryVideoProcFilterCaps.
+ * \code
+ * VAProcFilterCapSuperResolution sr_caps[VAProcSuperResolutionCount];
+ * unsigned int num_sr_caps = VAProcSuperResolutionCount;
+ * vaQueryVideoProcFilterCaps(va_dpy, vpp_ctx,
+ *     VAProcFilterSuperResolution,
+ *     &sr_caps, &num_sr_caps);
+ */
+typedef struct _VAProcFilterCapSuperResolution {
+    /** \brief Super Resolution mode, should be one of VAProcSuperResolutionXXX.*/
+    VAProcSuperResolutionMode            mode;
+    /** \brief The maximum value of input width SR filter is capable of supporting.*/
+    uint32_t                             max_input_width;
+    /** \brief The maximum value of input height SR filter is capable of supporting.*/
+    uint32_t                             max_input_height;
```

Comment:
how about max output resolution , for example: I have a 720P input , but suppose the capability of HW maybe just support 1080P output, dont support 4K, how to describe the restriction?

Reply:
Super resolution scaling itself supports 2x. Hence, only max input is meaningful. Probably, need to document this information. And also consider Jonathan's comment.